### PR TITLE
Adds actions on states

### PR DESCRIPTION
**Pull request by victorcotap**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,12 @@ Generate a JSON representation of Twine flows
   },
   "screen3": {
     "text": "Bluh bluh bluh",
-    "name": "Sorry"
+    "name": "Sorry",
+    "actions" : [
+        {"actionSequenceType": "pre", "actionType" : "saveItem", "values" : ["item_name", "photos"], }
```
Comment (marcolz) [inline README.md]: Invalid JSON.

File: index.js [hunk 2]
```diff
@@ -6,7 +8,7 @@ if (process.argv.length <= 2) {
   process.exit(-1)
 }
 
-const filename = process.argv[2]
+let filename = process.argv[2]
```
Comment (marcolz) [inline index.js]: ?

File: lib/yarn.js [hunk 3]
```diff
@@ -32,6 +34,29 @@ const yarn = {
...
+
+          action.actionSequenceType = elements[0]
+          action.actionType = elements[1]
+          action.values = values
+
+          passages[pid].actions.push(action)
+        })
+
+        text = text.replace(/{{.+}}/g, '').trim()
+
```
Comment (marcolz) [inline lib/yarn.js]: Unnecessary new line.

File: lib/yarn.js [hunk 4]
```diff
@@ -32,6 +34,29 @@ const yarn = {
...
+          const action = {}
+          const elements = actionText.split(':')
+          const values = elements[2].split('|')
+
+          action.actionSequenceType = elements[0]
+          action.actionType = elements[1]
+          action.values = values
+
+          passages[pid].actions.push(action)
+        })
```
Comment (marcolz) [inline lib/yarn.js]: This is a nice opportunity to use `reduce()` instead of `forEach()`.

File: lib/yarn.js [hunk 5]
```diff
@@ -32,6 +34,29 @@ const yarn = {
...
       })
 
+      const actions = text.match(/{{.+}}/g)
+      if (actions) {
+        actions.forEach((actionText) => {
+          actionText = actionText.replace(/^{{/, '').replace(/}}$/, '')
+
+          const action = {}
+          const elements = actionText.split(':')
+          const values = elements[2].split('|')
```
Comment (marcolz) [inline lib/yarn.js]: You could extract all these values with a single RegExp and deconstruct them straight into a Object.

File: lib/yarn.js [hunk 6]
```diff
@@ -32,6 +34,29 @@ const yarn = {
...
+          passages[pid].actions.push(action)
+        })
+
+        text = text.replace(/{{.+}}/g, '').trim()
+
+      }
+      passages[pid].text = text
+
+
+
```
Comment (marcolz) [inline lib/yarn.js:49]: One more?

**Comment by marcolz** [comment]
Left a few comments. LGTM!

**Comment by victorcotap** [comment]
updated with comments addressed

**Pull request merged**